src/seedu/addressbook/ui/OutputFormatter.java

```diff
@@ -0,0 +1,102 @@
+package seedu.addressbook.ui;
+
+import java.util.List;
+
+public abstract class OutputFormatter {
+    /**
+     * 
+     * @return Divider for sections of output
+     */
+    abstract String getDivider();
+
+    /**
+     * 
+     * @return A decorative prefix added to the beginning of lines
+     */
+    abstract String getLinePrefix();
+
+    /**
+     * 
+     * @return Regex eg. "\t%1$d. %2$s"
+     */
+    abstract String getIndexedListItemFormat();
+
+    /**
+     * 
+     * @return line separator
+     */
+    abstract String getLineSeparator();
+
+    /**
+     * 
+     * @param str
+     * @return A formatted line
+     */
+    public String formatLine(String str) {
+        return getLinePrefix() + str;
+    }
+
+    /**
+     * 
+     * @param message
+     * @return entire String formed by joining strings in message
+     */
+    // TODO use String builder
+    public String formatLines(String... message) {
+        String ret = "";
+        for (int i = 0; i < message.length; i++) {
+            String m = message[i];
+            ret += getLinePrefix() + m.replace("\n", "\n" + getLinePrefix())
+                    + (i < message.length - 1 ? getLineSeparator() : "");
+        }
+        return ret;
+    }
+
+    /**
+     * 
```

Comment:
it is better to follow the suggested guidelines (from the coding standard) to write header comments.
this one here in particular has a succinct description missing,